File: dev-docs/snippets/model-surface/query.swift
import JsBaoClient

// Mongo-style filtered query. The static `Task.query` returns the hydrated
// rows (`[Task]`) directly — there is no PaginatedResult wrapper. Reads span
// every open document by default.
func query() {
  // #region example
  let rows = Task.query(["priority": ["$gte": 2], "completed": false])
  // #endregion example
  _ = rows
}

File: dev-docs/snippets/model-surface/query.ts
import { Task } from "../../../examples/_harness/generated/ts/Task.generated";

// Mongo-style filtered query. Returns a PaginatedResult — the rows are on
// `.data`, with `.nextCursor` / `.hasMore` for pagination.
export async function query() {
  // #region example
  const result = await Task.query({ priority: { $gte: 2 }, completed: false });
  const rows = result.data;
  // #endregion example
  return rows;
}
